### PR TITLE
[#4035] Update ASF headers in all files

### PR DESCRIPTION
**Pull request by justinmclean**
### What changes were proposed in this pull request?

Update all Datastrato headers to ASF headers, including the short version in documents and generated headers.

### Why are the changes needed?

To comply with ASF policy.

Fix: #4035

### Does this PR introduce _any_ user-facing change?

Updated headers and a few lines of code that generated headers. increated the limit of length of python files as one test file was now over the limit with the bigger header.

### How was this patch tested?

Build and non-integration tests run locally all passed.

### CODE REVIEW

File: build.gradle.kts [hunk 1]
```diff
@@ -459,6 +459,7 @@ tasks.rat {
     "**/licenses/*.txt",
     "**/licenses/*.md",
     "integration-test/**",
+    "docs/*.md",
```
Comment (jerryshao) [inline build.gradle.kts]: I think we have to fix the license header for docs, right? Maybe not in this PR, can you please create an issue to track this?

Comment (jerryshao) [reply build.gradle.kts]: But it also includes like "Copyright 2023 Datastrato Pvt Ltd", is it OK?

Comment (justinmclean) [reply build.gradle.kts]: See #4050

Comment (justinmclean) [reply build.gradle.kts]: I also removed all "Copyright 2023 Datastrato Pvt Ltd". While most of the content is still copyright Datastrato, ASF policy is to mention that in the NOTICE file (known as a relocated copyright statement).

**Comment by jerryshao** [comment]
Let's merge it first, if there's any missing, we can fix it later on. This PR is too big and easy to conflict.

**Pull request merged**